src/diffusers/loaders.py

```diff
@@ -67,6 +67,36 @@
 CUSTOM_DIFFUSION_WEIGHT_NAME_SAFE = "pytorch_custom_diffusion_weights.safetensors"
 
 
+class PatchedLoraProjection(nn.Module):
+    def __init__(self, regular_linear_layer, lora_linear_layer, lora_scale=1):
+        super().__init__()
+        self.regular_linear_layer = regular_linear_layer
+        self.lora_linear_layer = lora_linear_layer
+        self.lora_scale = lora_scale
+
+    def forward(self, input):
+        return self.regular_linear_layer(input) + self.lora_scale * self.lora_linear_layer(input)
+
+
+def text_encoder_attn_modules(text_encoder):
+    attn_modules = []
+
+    if isinstance(text_encoder, CLIPTextModel):
+        for i, layer in enumerate(text_encoder.text_model.encoder.layers):
+            name = f"text_model.encoder.layers.{i}.self_attn"
+            mod = layer.self_attn
+            attn_modules.append((name, mod))
+    elif isinstance(text_encoder, T5EncoderModel):
+        for i, block in enumerate(text_encoder.encoder.block):
+            name = f"encoder.block.{i}.layer.0.SelfAttention"
+            mod = block.layer[0].SelfAttention
+            attn_modules.append((name, mod))
+    else:
+        raise ValueError(f"do not know how to get attention modules for: {text_encoder.__class__.__name__}")
```

Comment:
Needs to be a bit standardized IMO because we're throwing this from a library and not from a training script.